dev-tools/iota-private-network/experiments/network-fuzz-test.sh

```diff
@@ -142,66 +189,124 @@ iptables_unblock_bidirectional() {
+        log "Injecting ${D1}ms±${J1}ms latency from $A to $B"
+        mark_pair "$A" "$B"
+        apply_latency "$A" "${D1}ms" "${J1}ms"
+        log "Injecting ${D2}ms±${J2}ms latency from $B to $A"
+        mark_pair "$B" "$A"
+        apply_latency "$B" "${D2}ms" "${J2}ms"
+      done
+    done
 
   #  For each validator pair A-B, randomly apply one of five blocking actions with 1/50 probability each
-  duration=$((RANDOM % 120 + 60))
+  log "######################################"
+  log "Start blocking traffic"
+  duration=$(( RANDOM % (MAX_DURATION - MIN_DURATION + 1) + MIN_DURATION ))
   for ((i=0; i<${#validators[@]}; i++)); do
     for ((j=i+1; j<${#validators[@]}; j++)); do
       A=${validators[i]}
       B=${validators[j]}
       # bidirectional block
-      if (( RANDOM % 50 == 0 )); then
+      if (( RANDOM % BLOCK_PROB == 0 )); then
         log "Blocking bidirectional traffic between $A and $B for ${duration}s"
         (iptables_block_bidirectional "$A" "$B"; sleep $duration; iptables_unblock_bidirectional "$A" "$B") &
       fi
       # outgoing from A to B
-      if (( RANDOM % 50 == 1 )); then
+      if (( RANDOM % BLOCK_PROB == 1 )); then
         log "Blocking outgoing traffic from $A to $B for ${duration}s"
         (iptables_block "$A" "$B"; sleep $duration; iptables_unblock "$A" "$B") &
       fi
       # incoming to A from B
-      if (( RANDOM % 50 == 2 )); then
+      if (( RANDOM % BLOCK_PROB == 2 )); then
         log "Blocking incoming traffic to $A from $B for ${duration}s"
         (iptables_block_incoming "$A" "$B"; sleep $duration; iptables_unblock_incoming "$A" "$B") &
       fi
       # outgoing from B to A
-      if (( RANDOM % 50 == 3 )); then
+      if (( RANDOM % BLOCK_PROB == 3 )); then
         log "Blocking outgoing traffic from $B to $A for ${duration}s"
         (iptables_block "$B" "$A"; sleep $duration; iptables_unblock "$B" "$A") &
       fi
       # incoming to B from A
-      if (( RANDOM % 50 == 4 )); then
+      if (( RANDOM % BLOCK_PROB == 4 )); then
         log "Blocking incoming traffic to $B from $A for ${duration}s"
         (iptables_block_incoming "$B" "$A"; sleep $duration; iptables_unblock_incoming "$B" "$A") &
       fi
     done
   done
-
+  log "######################################"
+  log "Start stopping validators and apply netem loss"
   # Loop through validators
     for v in "${validators[@]}"; do
-      duration=$((RANDOM % 120 + 60)) # between 120 and 180 seconds
+      duration=$(( RANDOM % (MAX_DURATION - MIN_DURATION + 1) + MIN_DURATION )) # between MIN_DURATION and MAX_DURATION seconds
       loss=$((RANDOM % 41 + 10))       # 10–50% loss
-      r=$((RANDOM % 100))
-        if   (( r < 10 )); then
+      r=$(( RANDOM % 100 ))
+        if (( r < STOP_PROB )); then
           log "Stopping $v for ${duration}s"
           (restart_validator "$v" "$duration") &
-        elif (( r < 25 )); then
+        elif (( r < LOSS_PROB )); then
```

Comment:
Here, the probability of loss is LOSS_PROB-STOP_PROB
```suggestion
        elif (( r < LOSS_PROB + STOP_PROB )); then
```